app/logic/updateTrackHours.py

```diff
@@ -0,0 +1,75 @@
+# from app.controllers.admin import admin_bp
+from app.models.eventParticipant import EventParticipant
+from datetime import datetime
+from app.models.user import User
+
+
+def getEventLengthInHours(startTime, endTime, eventDate):
+    """
+    Converts the event length hours into decimal
+    parameters: startTime- start time event (type: time)
+                endTime- end time event (type: time)
+                eventDate- date of the event (type: datetime)
+    """
+    #can only subtract datetime objects, not time objects. So convert time into datetime
+    eventLength = datetime.combine(eventDate, endTime) - datetime.combine(eventDate, startTime)
+    (h, m, s) = str(eventLength).split(':')
+    eventLengthInHours = int(h) + round(int(m) / 60, 2)
+
+    return eventLengthInHours
+
+
+def updateTrackHours(participantData):
+    """
+    updates the events participant table with data from the track hours webpage
+
+    param: participantData- a dictionary that contains data from every row of the page along with the associated username.
+    """
+
+    for user in range(1, len(participantData)):
+        if f'username{user}' in participantData:
+            username = participantData[f'username{user}']
+            if (User.select().where(User.username == username)).exists():
+
+                try:
+                    if participantData['checkbox_'+ username] == "on": #if the user is marked as present

```

Comment:
What if there is no EventParticipant entry for the user here? Do you want to say they don't exist, or should we add them?

Reply:
We will want to say "Volunteer does not exist."